Formula/ehco.rb

```diff
@@ -0,0 +1,17 @@
+class Ehco < Formula
+  desc "Network relay tool and a typo :)"
+  homepage "https://github.com/Ehco1996/ehco"
+  url "https://github.com/Ehco1996/ehco/archive/refs/tags/v1.0.6.tar.gz"
+  sha256 "326c468c3790ad01031e52ccb4efcfa5e331d2198fdb13137749c67e8eaacf38"
+  license "AGPL-3.0-or-later"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-o", bin/"ehco", "cmd/ehco/main.go"
+  end
+
+  test do
+    shell_output("#{bin}/ehco -v")
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
okey, i wiil do it later,  thanks for you help.

Reply:
Sure. Please squash further changes with your first commit, and use the commit message
```
ehco 1.0.6 (new formula)
```
You'll also need to address the CI failure:
```
ehco:
  * Formula license ["AGPL-3.0-or-later"] does not match GitHub license ["GPL-3.0"].
Error: 1 problem in 1 formula detected
```

Reply:
got it.

Reply:
all ci check passed , please cc.

Reply:
We still need a test that does more than check the version.

Reply:
@carlocab 

add `--help` test part

but `ehco` is a network traffic relay server ,i don't  konw how to test it man feature in homebrew test.
there is a cmd `make test` in  source code, could you tell me how to run the test ? not familiar with ruby

Reply:
Sorry, but from my first comment:

>Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

Look for formulae that use `free_port` in their test blocks; I think those might be helpful. Otherwise, we can try a command that we know should fail with a specific exit code and check that we get the expected error message and exit code.

The source code isn't available in the `test` block, so we can't do `make test`. We also don't like unit tests as tests. Please read the documentation on adding test blocks for more information.